### PR TITLE
Fix wrong ksceKernelGetModuleList2

### PR DESCRIPTION
**Pull request by Princess-of-Sleeping**

### CODE REVIEW

**Comment by CelesteBlue-dev** [comment]
DONT MERGE THIS COMMIT
there are issues that must be discussed

**Comment by CelesteBlue-dev** [comment]
I am checking the ModuleMgr structs. It seams strange to have so many similar different structs.

**Comment by yne** [comment]
@CelesteBlue-dev : I would say it for the legacy handling.
Just like the PSP ModuleMgr, which became a mess year after year, so it can handle ELF/PRX from 1.x 2.x ... SDK.
I Guess they real struct name was `SceKernelModuleListInfo_1_00`, `SceKernelModuleListInfo_2_40`, `SceKernelModuleListInfo_3_60` ...

**Comment by yne** [comment]
Its was just a numbering example,
I don't really have the list of firmware/SDK version in which the ELF layout was changed.

**Comment by Princess-of-Sleeping** [comment]
Fixed to use union structure

**Comment by d3m3vilurr** [comment]
@Princess-of-Sleeping is this review still in WIP state?

**Comment by Princess-of-Sleeping** [comment]
@d3m3vilurr I did not set it in the WIP state. @CelesteBlue-dev has been set to WIP state.

**Comment by d3m3vilurr** [comment]
@Princess-of-Sleeping it looks good to me. @yne can you review this?

File: include/psp2kern/kernel/modulemgr.h [hunk 1]
```diff
@@ -157,20 +168,34 @@ int ksceKernelSearchModuleByName(const char *module_name, const char *path, int
...
- * @param[out] infolists - infolists output
- * @param[in] num - Specify the maximum number of modinfolist to retrieve. If the function returns 0, it returns the number of modules loaded in the target pid in num
+ * @param[in]    pid       - target pid
+ * @param[out]   infolists - infolists output
+ * @param[inout] num       - Specify the maximum number of modinfolist to retrieve. If the function returns 0, it returns the number of modules loaded in the target pid in num
  *
  * @return 0 on success, < 0 on error.
  */
-int ksceKernelGetModuleList2(SceUID pid, SceKernelLoadedModuleInfo *infolists, size_t *num);
+int ksceKernelGetModuleList2(SceUID pid, SceKernelModuleListInfo *infolists, size_t *num);
```
Comment (d3m3vilurr) [inline include/psp2kern/kernel/modulemgr.h]: but param type was changed :)

**Comment by yne** [comment]
I don't like the fact that the formating is not respected (`union{` instead of `union {`) but I don't care anymore because that the kind of reject you shall get from a linter (if this project could have one) instead of a human.

**Comment by d3m3vilurr** [comment]
@yne you right.
@Princess-of-Sleeping can you change the formatting of `struct{` and `union{`?

**Comment by Princess-of-Sleeping** [comment]
changed the formatting

**Comment by d3m3vilurr** [comment]
LGTM. it should change to hardtab but I'll recheck whole codeset in later.
thank you for your investigation and patient.

**Pull request merged**